src/thx/schema/macro/Arguments.hx

```diff
@@ -0,0 +1,103 @@
+package thx.schema.macro;
+
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import thx.schema.macro.Error.*;
+import haxe.ds.Option;
+
+class Arguments {
+  /**
+   *  Generates a schema from the type passed as the first argument.
+   *  @param typeRef
```

Comment:
No it is actually correct. Optional arguments are not easy in macro land and it is easier to check the arity yourself by using the magic `Array<Expr>`